bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+function getCIPrNumber(): number | null {
+  if (process.env.GITHUB_REF_NAME?.endsWith('/merge')) {
+    // https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+    return Number(process.env.GITHUB_REF_NAME.split('/')[0]);
+  } else if (process.env.CIRCLE_PULL_REQUEST?.includes('/pull/')) {
+    // https://circleci.com/docs/variables/#built-in-environment-variables
+    return Number(process.env.CIRCLE_PULL_REQUEST.split('/').at(-1));
+  } else {
+    return null;
+  }
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer;
+
+// If you change this, you might want to update the one in the website transformer
+// TODO: Change this when GH_TOKEN isn't needed to fetch PR release versions anymore
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  try {
+    if (_allPrReleaseVersions) {
+      return _allPrReleaseVersions;
+    }
+
+    if (process.env.NODE_ENV === 'test') {
+      const versions: PrReleaseVersionsContainer = {
+        22533: {
+          release: '',
+          backports: [] as string[],
+        },
+        26789: {
+          release: '',
+          backports: [] as string[],
+        },
+        37094: {
+          release: '',
+          backports: [] as string[],
+        },
+      };
+
+      _allPrReleaseVersions = versions;
+
+      const ciPrNumber = getCIPrNumber();
+      if (ciPrNumber) {
+        console.log(
+          'Detected PR number:\n\n' +
+            `'${ciPrNumber}'\n\n` +
+            'using CI environment variables. Adding to list of PRs...\n',
+        );
+        _allPrReleaseVersions[ciPrNumber] = {
+          release: '',
+          backports: [],
+        };
+      }
+
+      return _allPrReleaseVersions;
+    }
+
+    if (!process.env.GH_TOKEN) {
+      throw new Error(`GH_TOKEN is required for fetching PR release versions.`);
+    }
+
+    const fetchOptions = {
+      method: 'GET',
+      headers: {
+        Accept: 'application/vnd.github+json',
+        'X-GitHub-Api-Version': '2022-11-28',
+        Authorization: `Bearer ${process.env.GH_TOKEN}`,
+      },
+    };
+
+    const artifactsListResponse = await fetch(
+      'https://api.github.com/repos/electron/website/actions/artifacts',
+      fetchOptions,
+    );
+    const latestArtifact = (await artifactsListResponse.json()).artifacts
+      .filter(({ name }: { name: string }) => name === 'resolved-pr-versions')
+      .sort((a: { id: number }, b: { id: number }) => a.id > b.id)[0];
+
+    const archiveDownloadResponse = await fetch(latestArtifact.archive_download_url, fetchOptions);
+    const buffer = Buffer.from(await archiveDownloadResponse.arrayBuffer());
+
+    const zip = new AdmZip(buffer);
+    const parsedData = JSON.parse(zip.readAsText(zip.getEntries()[0]!)) as PrReleaseArtifact;
```

Comment:
nit: Would it be possible to expand this line into a few more to make it easier to debug if something goes wrong?

For example:
* Checking `Array.isArray` and `length` instead of non-null coercing
* An `isPRReleaseArtifact` type guard function instead of a type assertion

Reply:
I just shamelessly copy pasted @dsanders11's code ¯\\_(ツ)_/¯ https://github.com/electron/website/pull/586#issue-2374053099

Will work on this in https://github.com/electron/website/pull/594/files#r1671021901